iOS_SDK/OneSignalSDK/Source/OneSignal.m

```diff
@@ -1928,6 +1928,8 @@ + (void)registerUserInternal {
                     callbackSet.failureBlock(error);
             }
         }
+        
+        [OSMessagingController.sharedInstance updateInAppMessagesFromCache];
```

Comment:
I'm confused with one thing since you are checking for user registration to show IAM. At this point, OneSignal.isRegisterUserFinished; will return true? because it will immediateOnSessionRetry, then both _registerUserFinished || isOnSessionSuccessfulForCurrentState will be false right? Am I missing something? Are the cached IAM being displayed? Thanks in advance for clarifying this

Reply:
Ya you are right this isn't working we would need a registerUserFailed boolean or settings registerUserFinished to true if it fails (it gets reset when it is called again).

Reply:
I added a registerUserSuccessful boolean to be used in place of registerUserFinished in some places. Added comments to make it clear that registerUserFinished is also set to true when the call fails